### PR TITLE
CMR-6604: Added support for service tab feature in collection preview.

### PR DESCRIPTION
**Pull request by sxu123**

### CODE REVIEW

File: collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj [hunk 1]
```diff
@@ -122,10 +123,32 @@
...
+                                     (search/find-concepts context :service)
+                                     :results)
+                items (:items (json/parse-string service-results true))]]
+      (first items))))
+
 (defn- get-additional-information
   "Retrieve additional information for collection rendering."
   [context concept-id]
-  {"granule_count" (get-granule-count-for-collection context concept-id)})
+  (println "services: " (map #(umm-json/umm->json %) (get-services-for-collection context concept-id)))
```
Comment (daniel-zamora) [inline collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj:148]: Should this println be here?

**Review by daniel-zamora — CHANGES_REQUESTED** [review]

File: collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj (hunk 1 above)
Comment (sxu123) [reply collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj]: Removed

**Review by daniel-zamora — APPROVED** [review]

File: collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj [hunk 2]
```diff
@@ -122,10 +123,31 @@
...
+        items (:items (json/parse-string collection-results true))
+        service-ids (get-in (first items) [:meta :associations :services])]
+    (for [service-id service-ids
+          :let [service-results (->> {:concept-id service-id
+                                      :result-format {:format :umm-json-results}}
+                                     (params/parse-parameter-query context :service)
+                                     (search/find-concepts context :service)
+                                     :results)
+                items (:items (json/parse-string service-results true))]]
+      (first items))))
```
Comment (mschmele) [inline collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj:143]: Nitpick: maybe add a comment about why you're only taking the first item

**Review by mschmele — COMMENTED** [review]
Code looks good - I'll approve once that question is answered

File: collection-renderer-lib/resources/collection_preview/collection_preview.erb [hunk 3]
```diff
@@ -6,6 +6,16 @@
 require 'json'
 metadata =  JSON.parse(umm_json)
 
+services = additional_information['services']
```
[the remark this answers is not in the document]
Comment (ryanmiller-1) [reply collection-renderer-lib/resources/collection_preview/collection_preview.erb:9]: The preview gem should expect consistent inputs from the applications that use it.  Both applications pass the collection metadata as a ruby hash, so passing the services' metadata as an array of hashes in the same way is consistent.  Additionally, the CMR was previously passing additional_information as a Java::ClojureLang::PersistentArrayMap. Passing it as a ruby hash here allows the preview gem to have more predictable outcomes for both applications.

**Review by mschmele — APPROVED** [review]

File: collection-renderer-lib/resources/collection_preview/collection_preview.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (mschmele) [reply collection-renderer-lib/resources/collection_preview/collection_preview.erb:9]: Sounds good - thanks!

File: collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj (hunk 2 above)
Comment (sxu123) [reply collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj:143]: One service-id will return at most one record.

**Pull request merged**